epistemix_platform/use_cases/write_to_local.py

```
@@ -0,0 +1,154 @@
+        except Exception as e:
+            raise IOError(f"Failed to write ZIP archive to {file_path}: {e}")
+
+    elif content.content_type == ContentType.BINARY:
+        # For binary content, raw_content should be hex-encoded data
+        if content.encoding == "hex":
+            # Validate and write hex-encoded content as binary
+            if not _validate_hex(content.raw_content):
+                raise ValueError(
+                    f"Binary content is not valid hex: content starts with "
+                    f"'{content.raw_content[:50]}...'"
+                )
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)